DscResources/MSFT_xMySqlDatabase/MSFT_xMySqlDatabase.schema.mof

```diff
@@ -1,9 +1,10 @@
 [ClassVersion("1.0.0"),FriendlyName("xMySqlDatabase")] 
 class MSFT_xMySqlDatabase : OMI_BaseResource
 {
-    [Key, Description("Name of the database.")] String Name;
-    [Required, Description("Should the database be present or absent."), ValueMap{"Present","Absent"}, Values{"Present","Absent"}] String Ensure;
-    [Required, Description("The root credential that is used to install MySql server."), EmbeddedInstance("MSFT_Credential")] String ConnectionCredential;
+    [Key, Description("Name of the database.")] String DatabaseName;
```

Comment:
Sure this will break the existing users of this, but personally I think this is better. It consitently names all references to the database regardless of resource being referenced. So all references to the database would end up being DatabaseName, name while generic could refer to any number of things in any of the resources.